felix/cmd/calico-bpf/commands/conntrack.go

```diff
@@ -424,46 +443,36 @@ func runClean(c *cobra.Command, _ []string) {
 
 type conntrackCreateCmd struct {
 	*cobra.Command
-	Version string `docopt:"--ver"`
-	version string
+	conntrackOpts
 }
 
 func newConntrackCreateCmd() *cobra.Command {
 	cmd := &conntrackCreateCmd{
 		Command: &cobra.Command{
 			Use:   "create [--ver=<version>]",
-			Short: "create a conntrack map of specified version",
+			Short: "Creates a conntrack map of specified version",
 		},
 	}
 
-	cmd.Command.Flags().StringVarP((&cmd.version), "ver", "v", "", "conntrack version to create")
+	cmd.Command.Flags().IntVarP((&cmd.version), "ver", "v", 3, "conntrack version to create")
 	cmd.Command.Args = cmd.Args
 	cmd.Command.Run = cmd.Run
 
 	return cmd.Command
 }
 
-func (cmd *conntrackCreateCmd) Args(c *cobra.Command, args []string) error {
-	a, err := docopt.ParseArgs(makeDocUsage(c), args, "")
-	if err != nil {
-		return err
-	}
-
-	err = a.Bind(cmd)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
 func (cmd *conntrackCreateCmd) Run(c *cobra.Command, _ []string) {
 	var ctMap maps.Map
-	switch cmd.version {
-	case "2":
-		ctMap = conntrack.MapV2()
-	default:
-		ctMap = conntrack.Map()
+
+	cmd.ipv6 = ipv6 != nil && *ipv6
+
+	// Set the map size based on the actual max entries obtained from the map info.
+	if err := setCTMapSize(cmd.version, cmd.ipv6); err != nil {
+		log.WithError(err).Fatal("Failed to set ConntrackMap size")
 	}
+
+	ctMap = getCTMap(cmd.version, cmd.ipv6)
+
 	if err := ctMap.EnsureExists(); err != nil {
 		log.WithError(err).Errorf("Failed to create conntrackMap version %s", cmd.version)
```

Comment:
govet fails here because cmd.version is string

```
cmd/calico-bpf/commands/conntrack.go:477:3: printf: (*github.com/sirupsen/logrus.Entry).Errorf format %s has arg cmd.version of wrong type int (govet)04:43
		log.WithError(err).Errorf("Failed to create conntrackMap version %s", cmd.version)04:43
```

Reply:
good to run `make static-checks` before pushing a commit

Reply:
Oops, sorry for that ~